### PR TITLE
DEPR: removed deprecated argument `obj` from GroupBy `get_group`

### PR DESCRIPTION
**Pull request by natmokval**
- xref #53571
- removed deprecated argument `obj` from GroupBy `get_group`
- added an entry in the doc/source/whatsnew/v3.0.0.rst

### CODE REVIEW

**Comment by natmokval** [comment]
@MarcoGorelli, could you please take a look at this PR? CI failures aren't related to my changes.

File: pandas/core/groupby/groupby.py [hunk 1]
```diff
@@ -1146,14 +1145,7 @@ def get_group(self, name, obj=None) -> DataFrame | Series:
...
         else:
-            warnings.warn(
-                "obj is deprecated and will be removed in a future version. "
-                "Do ``df.iloc[gb.indices.get(name)]`` "
-                "instead of ``gb.get_group(name, obj=df)``.",
-                FutureWarning,
-                stacklevel=find_stack_level(),
-            )
-            return obj._take_with_is_copy(inds, axis=self.axis)
+            raise ValueError("cannot pass argument obj to get_group")
```
Comment (WillAyd) [inline pandas/core/groupby/groupby.py:1148]: Sorry if I missed any conversations on this but I am fairly certain you can just remove the `obj=None` parameter from the function definition and then remove this `if...else` block altogether

Comment (rhshadrach) [reply pandas/core/groupby/groupby.py:1148]: Agreed

Comment (natmokval) [reply pandas/core/groupby/groupby.py]: thank you, I removed the parameter `obj` from the definition of `get_group`, the block `if...else` and corrected docstring.

File: doc/source/whatsnew/v3.0.0.rst [hunk 2]
```diff
@@ -96,6 +96,13 @@ Deprecations
 - Deprecated :meth:`Timestamp.utcnow`, use ``Timestamp.now("UTC")`` instead (:issue:`56680`)
 -
 
+.. ---------------------------------------------------------------------------
+.. _whatsnew_300.prior_deprecations:
+
+Removal of prior version deprecations/changes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+- Removed deprecated argument ``obj`` in :meth:`GroupBy.get_group` (:issue:`53545`)
```
Comment (rhshadrach) [inline doc/source/whatsnew/v3.0.0.rst:104]: In order to resolve a link to the documentation, can you replace the end with

    :meth:`DataFrameGroupBy.get_group` and :meth:`SeriesGroupBy.get_group`

**Review by rhshadrach — CHANGES_REQUESTED** [review]
Looks good!

File: doc/source/whatsnew/v3.0.0.rst (hunk 2 above)
Comment (natmokval) [reply doc/source/whatsnew/v3.0.0.rst]: thank you, I corrected the note in v3.0.0.rst

**Review by WillAyd — APPROVED** [review]

**Review by rhshadrach — APPROVED** [review]
lgtm

**Comment by rhshadrach** [comment]
Thanks @natmokval!

**Pull request merged**